### DISCUSSION TITLE
Trying to run the actions in the Create 3 tutorial

### DISCUSSION THREAD

**Discussion by cpeavy2** [iRobotEducation/create3_docs#467] [Questions] [answered]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Dell (i7-4500U CPU), Ubuntu 22.04.3 LTS

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.2.1

### Which RMW is your robot running?

CycloneDDS

### Does your robot have an assigned namespace? If so, please share.

No

### Is the robot connected to a network? If so, what is the network type?

2.4GHz Wi-Fi

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

Yes

### What is the Adapter Board's USB/BLE Toggle currently switched to?

Bluetooth (default)

### Describe your question.

I'm trying to run the actions in the Create 3 tutorial https://iroboteducation.github.io/create3_docs/api/docking/ . Docking and undocking for example. When I run:

ros2 action send_goal /undock irobot_create_msgs/action/Undock "{}"

The system replies:

ros2 action send_goal /undock irobot_create_msgs/action/Undock "{}"
Waiting for an action server to become available...
Sending goal:
     {}

And the robot just sits there, does not move. I can see the "/undock" action server and I can drive it around with teleop_twist_keyboard so it's seeing /cmd_vel.

Same thing with Drive Goals (/drive_arc and /drive_distance) and Wall Follow (/wall_follow).

**Comment by alsora** [comment, marked as answer]
Hi, the issue seems to indicate that the laptop is sending the goal request, but the robot is not reacting to it.

First of all, I would recommend to verify that robot and laptop are using the same RMW.
You said that the robot is using cyclonedds as RMW, note that the default RMW for ROS 2 Humble is fast-dds, so make sure that you changed it on the laptop.

To check and modify RMW on the robot see: https://iroboteducation.github.io/create3_docs/webserver/logs/
To check and modify RMW on the laptop see: https://iroboteducation.github.io/create3_docs/setup/xml-config/

If you still have the issue after you have verified that both RMWs are the same, could you please share with us a log from the robot?
It can be found in the webserver https://iroboteducation.github.io/create3_docs/webserver/logs/

**Reply by cpeavy2** [reply]
Thanks! Changing the robot to fast-dds solved the problem. The robot now moves, yay!   :- ]
